tests/pipelines/test_train_step.py

```diff
@@ -160,3 +213,51 @@ def test_train_steps_with_correct_tags(tmp_pipeline_root_path):
     assert tags["mlflow.pipeline.template.name"] == "regression/v1"
     assert tags["mlflow.pipeline.step.name"] == "train"
     assert tags["mlflow.pipeline.profile.name"] == "test_profile"
+
+
+def test_train_step_with_tuning_best_parameters(tmp_pipeline_root_path):
+    with mock.patch.dict(
+        os.environ, {_MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR: str(tmp_pipeline_root_path)}
+    ):
+        train_step_output_dir = setup_train_dataset(tmp_pipeline_root_path)
+        train_step = setup_train_step(tmp_pipeline_root_path, use_tuning=True)
+        train_step._run(str(train_step_output_dir))
+    assert (train_step_output_dir / "best_parameters.yaml").exists()
+
+
+def test_train_step_with_tuning_child_runs(tmp_pipeline_root_path):
+    with mock.patch.dict(
+        os.environ, {_MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR: str(tmp_pipeline_root_path)}
+    ):
+        train_step_output_dir = setup_train_dataset(tmp_pipeline_root_path)
+        train_step = setup_train_step(tmp_pipeline_root_path, use_tuning=True)
+        train_step._run(str(train_step_output_dir))
+
+    with open(train_step_output_dir / "run_id") as f:
+        run_id = f.read()
+
+    run = MlflowClient().get_run(run_id)
+    child_runs = train_step._get_tuning_df(run)
+    assert len(child_runs) == 3
```

Comment:
Can we assert that the expected params / metric values are logged to the child runs?